scripts/init.d/cgconfig.in

```diff
@@ -81,60 +98,62 @@ create_default_groups() {
         # special rule for cpusets
         #
         if echo $controllers | grep -q -w cpuset; then
-                cpus=`cgget -nv -r cpuset.cpus /`
+                cpus=$(cgget -nv -r cpuset.cpus /)
                 cgset -r cpuset.cpus=$cpus $defaultcgroup
-                mems=`cgget -nv -r cpuset.mems /`
+                mems=$(cgget -nv -r cpuset.mems /)
                 cgset -r cpuset.mems=$mems $defaultcgroup
         fi
 
         #
         # Classify everything to default cgroup. Ignore errors, some processes
         # may exit after ps is run and before cgclassify moves them.
         #
-        cgclassify -g $controllers:$defaultcgroup `ps --no-headers -eL o tid` \
+        cgclassify -g $controllers:$defaultcgroup $(ps --no-headers -eL o tid) \
                  2>/dev/null || :
 }
 
 start() {
-        echo -n "Starting cgconfig service: "
-	if [ -f "$lockfile" ]; then
+        printf "Starting cgconfig service: "
+	if [[ -f "$lockfile" ]]; then
             log_warning_msg "lock file already exists"
             return 0
         fi
 
-        if [ $? -eq 0 ]; then
-                if [ ! -s $CONFIG_FILE ]; then
-                    log_failure_msg $CONFIG_FILE "is not configured"
-                    return 6
-                fi
-
-                $CGCONFIGPARSER_BIN -l $CONFIG_FILE
-                retval=$?
-                if [ $retval -ne 0 ]; then
-                    log_failure_msg "Failed to parse " $CONFIG_FILE
-                    return 1
-                fi
+        if [[ ! -s "$CONFIG_FILE" ]]; then
+          log_failure_msg $CONFIG_FILE "is not configured"
+          return 6
+        fi
+
+
+        if ! "$CGCONFIGPARSER_BIN" -l "$CONFIG_FILE" -L "$CONFIG_DIR"
+        then
+          log_failure_msg "Failed to parse " "$CONFIG_FILE" "or" "$CONFIG_DIR"'/*'
+          return 1
         fi
 
         if [ $CREATE_DEFAULT = "yes" ]; then
                 create_default_groups
         fi
 
-        touch "$lockfile"
-        retval=$?
-        if [ $retval -ne 0 ]; then
+        if ! mkdir -p "$lockfiledir" ; then
+          log_failure_msg "Failed to mkdir $lockfiledir directory"
+          return 1
+        fi
+
+
+        if ! touch "$lockfile" ; then
             log_failure_msg "Failed to touch $lockfile"
             return 1
         fi
-        log_success_msg
+        log_success_msg "Started $servicename"
         return 0
 }
 
 stop() {
-    echo -n "Stopping cgconfig service: "
-    rm -f "$lockfile"
-    log_success_msg
-    return 0
+    echo -n "Stopping cgconfig service is not supported!: "
+    # rm -f "$lockfile"
```

Comment:
I guess you can remove the `rm -f $lockfile`.  The echo can be replaced with print, here too.

Reply:
Done